[supervisord]
logfile_maxbytes=1MB
logfile_backups=2
nodaemon=true

[eventlistener:dependent-startup]
command=python3 -m supervisord_dependent_startup
autostart=true
autorestart=unexpected
startretries=0
exitcodes=0,3
events=PROCESS_STATE
buffer_size=1024

[program:rsyslogd]
command=/usr/sbin/rsyslogd -n -iNONE
priority=1
autostart=false
autorestart=false
stdout_logfile=syslog
stderr_logfile=syslog
dependent_startup=true

[program:veth-creator]
command=/usr/bin/veth-create.sh
priority=2
startsecs=0
autostart=false
autorestart=false
dependent_startup=true

[program:redis]
command=/usr/bin/redis_start.sh --bind 127.0.0.1 --port 6379
priority=3
autostart=false
autorestart=true
stdout_logfile=syslog
stderr_logfile=syslog
dependent_startup=true
dependent_startup_wait_for=rsyslogd:running

[program:redis-cmd-listener]
command=/bin/bash -c "python3 /sai-challenger/scripts/redis-cmd-listener.py"
priority=4
autostart=false
autorestart=true
stdout_logfile=syslog
stderr_logfile=syslog
dependent_startup=true
dependent_startup_wait_for=rsyslogd:running redis:running

[program:syncd]
command=/usr/bin/syncd -s -p /etc/sai.d/sai.profile
priority=5
autostart=false
autorestart=true
stdout_logfile=syslog
stderr_logfile=syslog
dependent_startup=true
dependent_startup_wait_for=rsyslogd:running redis:running veth-creator:exited

